extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/user/UserService.java

```diff
@@ -259,8 +263,13 @@ private String generateLDAPQuery(String username)
 
             // Add all DNs for found users
             while (results.hasMore()) {
-                LDAPEntry entry = results.next();
-                userDNs.add(entry.getDN());
+                try {
+                    LDAPEntry entry = results.next();
+                    userDNs.add(entry.getDN());
+                } catch (LDAPReferralException e) {
+                    logger.debug("Ignoring LDAP Referral: \"{}\".", e.toString());
```

Comment:
Fair enough - my thought process was that at present, it is simply not possible to log in via LDAP if a referral is returned. This makes it impossible to use LDAP auth against an AD where the search base dn is the root of the AD, because at present the LDAPReferralException will be rethrown and not dealt with in any way. This was a v0.1 attempt to at least improve the situation a little. I don't disagree that including the capability to (probably optionally) follow the referral is a better solution, but I'm afraid I'll have to leave that to someone more familiar with the code, then.